app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,90 @@
+class OrderItemsController < ApplicationController
+  before_action :check_for_duplicate_product, only: [:create]
+
+  def create
+    @order = current_order
```

Comment:
I think this same line is used in a few different places in these controller methods so it would be a good opportunity to use a `before_action`